codegen/module_system.go

```diff
@@ -544,6 +567,121 @@ func (g *httpClientGenerator) Generate(
 	}, nil
 }
 
+// PopulateQPSLevels loops through endpoint dir and gets qps levels
+func PopulateQPSLevels(EndpointsBaseDir string) (map[string]int, error) {
+	qpsLevels := make(map[string]int)
+	filesList := []string{}
+	endpointFiles, err := GetListOfAllFilesInEndpointDir(EndpointsBaseDir, filesList)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"Error in getting endpoint files",
+		)
+	}
+	for _, endpointFile := range endpointFiles {
+		config, err := UnMarshalEndpointFile(endpointFile)
+		if err != nil {
+			return nil, errors.Wrapf(
+				err,
+				"Error in unmarshalling endpoint file %q",
+				endpointFile,
+			)
+		}
+		clientMethod, methodOK := config["clientMethod"]
+		clientID, clientOK := config["clientId"]
+		qpsLevel, qpsOK := config["qpsLevel"]
+		// these fields exist in the config files
+		hasFields := methodOK && clientOK && qpsOK
+		if hasFields {
+			// when yaml files have no values for these keys(fields)
+			clientless := clientID == nil || clientID == ""
+			methodless := clientMethod == nil || clientMethod == ""
+			if !clientless && !methodless {
+				// unique key because of potential clients having same method names (staging)
+				key := clientID.(string) + "-" + clientMethod.(string)
+				// store highest qps level for circuit breaker in qpsLevels map
+				thisQPSLevel := int(qpsLevel.(float64))
+				if currentQPSLevel, ok := qpsLevels[key]; ok {
+					if thisQPSLevel > currentQPSLevel {
+						qpsLevels[key] = thisQPSLevel
+					}
+				} else {
+					qpsLevels[key] = thisQPSLevel
+				}
+			}
+		}
+	}
+	return qpsLevels, nil
+}
+
+// UnMarshalEndpointFile unmarshals endpoint file into config
+func UnMarshalEndpointFile(endpointFile string) (map[string]interface{}, error) {
+	var config map[string]interface{}
+	bytes, err := ioutil.ReadFile(endpointFile)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"Error in reading endpoint file %q",
+			endpointFile,
+		)
+	}
+	fileExtension := filepath.Ext(endpointFile)
+	if fileExtension == ".json" {
+		err = json.Unmarshal(bytes, &config)
+		if err != nil {
+			return nil, errors.Wrapf(
+				err,
+				"Error in unmarshalling json %q",
+				endpointFile,
+			)
+		}
+	}
+	if fileExtension == ".yaml" {
+		err = yaml.Unmarshal(bytes, &config)
+		if err != nil {
+			return nil, errors.Wrapf(
+				err,
+				"Error in unmarshalling yaml %q",
+				endpointFile,
+			)
+		}
+	}
+	return config, nil
+}
+
+// GetListOfAllFilesInEndpointDir gets all the endpoint config files
+// takes empty filesList array as parameter to add to during recursion
+func GetListOfAllFilesInEndpointDir(filePath string, filesList []string) ([]string, error) {
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return nil, errors.Errorf(
+			"Error in doing Stat with file path %q",
```

Comment:
Go error strings should not be capitalized: https://github.com/golang/go/wiki/CodeReviewComments#error-strings Can we make this "error" in all places? Thanks for adding error handling in all places